src/renderers/shared/stack/reconciler/ReactChildReconciler.js

```diff
@@ -50,12 +52,31 @@ var ReactChildReconciler = {
    * @return {?object} A set of child instances.
    * @internal
    */
-  instantiateChildren: function(nestedChildNodes, transaction, context) {
+  instantiateChildren: function(
+    nestedChildNodes,
+    transaction,
+    context,
+    selfDebugID // __DEV__ only
+  ) {
     if (nestedChildNodes == null) {
       return null;
     }
     var childInstances = {};
-    traverseAllChildren(nestedChildNodes, instantiateChild, childInstances);
+
+    if (__DEV__) {
+      traverseAllChildren(
+        nestedChildNodes,
+        (childInsts, child, name) => instantiateChild(
```

Comment:
I tried to put `selfDebugID` as the first arg of `instantiateChild` and to use `instantiateChild.bind(null, selfDebugID)` here, but that means we need to call `bind` as well in the production build. Do you think the current approach (arrowed currying, if that's a term) is acceptable? @spicyj